### DISCUSSION TITLE
How to change IPHONEOS_DEPLOYMENT_TARGET in Podsproject

### DISCUSSION THREAD

**Discussion by AbduShalata** [codemagic-ci-cd/codemagic-docs#2586] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

I have this error 

> /Users/builder/clone/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 11.0, but the range of supported deployment target versions is 12.0 to 17.4.99. (in target 'Flipper-Fmt' from project 'Pods')

How to change IPHONEOS_DEPLOYMENT_TARGET in pod

### Expected results

change IPHONEOS_DEPLOYMENT_TARGET

### Actual results

can't change it because i don't have xcode on my local machine

**Comment by dtrdic** [comment]
Hi @AbduShalata, locate the .`pbxproj` file and in there you can find the `IPHONEOS_DEPLOYMENT_TARGET` flag and set the minimum deployment target.
But the log you've sent is a warning not an error, please share a build id so i can check the logs

**Reply by AbduShalata** [reply]
Build Id = 6667f959eb60004a0221f13e ,I Need to change it in pod.pbxproj

**Reply by dtrdic** [reply]
please try adding the archive flag to your build command in yaml
```
    xcode-project build-ipa \
      --workspace "$CM_BUILD_DIR/ios/$XCODE_WORKSPACE" \
      --scheme "$XCODE_SCHEME" \
      --archive-flags="-destination 'generic/platform=iOS'"     
```

**Reply by AbduShalata** [reply]
Same Error  , Build Id = 66680f16f29cc70268872561

**Reply by dtrdic** [reply]
now i can see this error;
_`​no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?`_
Let's try the XCode suggestion, renaming the function, let's see if it resolves the failed build. Check also [this SO discussion ](https://stackoverflow.com/questions/77133361/no-template-named-unary-function-in-namespace-std-did-you-mean-unary-fun)for reference

**Reply by AbduShalata** [reply]
where can i find it ??,I don't have mac machine

**Reply by dtrdic** [reply]
Hmm one thing that you can do is access our VM-s when the build is running, whether by SSH or VNC connection, then locate the generated Podfile, copy it to your local machine and make the suggested modifications from SO discussion.
So;
- Start a build, enable SSH connection
- SSH or VNC into the machine
- When in the machine, navigate to /Users/builder/clone/ios
- There, a Podfile should be generated. Copy the contents of this Podfile, and create a Podfile locally inside your ios directory.
- Paste the contents of the VM-s Podfile to your locally created one.
- Now, inside the Podfile, add the removed libcpp function in your podfile as mentioned [here](https://stackoverflow.com/questions/77133361/no-template-named-unary-function-in-namespace-std-did-you-mean-unary-fun#:~:text=Until%20this%20is%20fixed%20in%20the%20source%20files%2C%20the%20best%20option%20is%20to%20add%20the%20removed%20libcpp%20function%20in%20your%20podfile%2C%20as%20solved%20here%20https%3A//github.com/facebook/react%2Dnative/issues/37748%23issuecomment%2D1580589448)

**Reply by AbduShalata** [reply]
give me same error, Build id :666931b708f2f0ccbbb5093a

**Reply by dtrdic** [reply]
The error logs now show permission denied errors when trying to execute this scripts during the build process.
```
Script-46EB2E000210E0.sh: line 2: /Users/builder/clone/ios/Pods/Target Support Files/OpenSSL-Universal/OpenSSL-Universal-xcframeworks.sh: Permission denied
Script-46EB2E000210D0.sh: line 2: /Users/builder/clone/ios/Pods/Target Support Files/Flipper-Glog/Flipper-Glog-xcframeworks.sh: Permission denied
Script-46EB2E000210C0.sh: line 2: /Users/builder/clone/ios/Pods/Target Support Files/Flipper-DoubleConversion/Flipper-DoubleConversion-xcframeworks.sh: Permission denied
```
Can you try by granting execution permission to the .sh script files located in the Target Support Files directory?
```
chmod +x /Users/builder/clone/ios/Pods/Target\ Support\ Files/OpenSSL-Universal/OpenSSL-Universal-xcframeworks.sh
chmod +x /Users/builder/clone/ios/Pods/Target\ Support\ Files/Flipper-Glog/Flipper-Glog-xcframeworks.sh
chmod +x /Users/builder/clone/ios/Pods/Target\ Support\ Files/Flipper-DoubleConversion/Flipper-DoubleConversion-xcframeworks.sh
```

**Reply by AbduShalata** [reply]
i did this and i had this error :

> CompileC /Users/builder/Library/Developer/Xcode/DerivedData/tijary-dafzycirwgblviekpxynbxzuwkus/Build/Intermediates.noindex/ArchiveIntermediates/tijary/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/ReactCommon.build/Objects-normal/arm64/TurboModuleUtils.o /Users/builder/clone/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.cpp normal arm64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'ReactCommon' from project 'Pods')

build id :66699f2304df255bc11e3569

**Reply by dtrdic** [reply]
the main error you have is this one now
`/Users/builder/clone/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.h:13:10: 'jsi/jsi.h' file not found`
Please check this [SO suggestion](https://stackoverflow.com/questions/76642961/jsi-jsi-h-file-not-found-error-when-upgrading-react-native-from-0-64-0-to-0) as a potential solution

**Reply by AbduShalata** [reply]
i know because this file create by   
> npm i

**Reply by dtrdic** [reply]
yes, try the Stack Overflow suggestion that I shared if it helps. Also, asking in the React community [here](https://github.com/react-native-community/upgrade-support/issues/240) would be better, as it's not a Codemagic-specific issue

**Comment by icarusdust** [comment]
To begin with, let me highlight that this is more about debugging your React Native project locally, and please note that you always need to make sure that your project works ok on your local machine before trying it with Codemagic. However, I’d like to give you some hints on how to resolve the issue and what is actually happening. 

**What is the issue about?** It can be reproduced locally by messing with the project pods but hard to say anything because your project has completely different settings and I cannot see what exactly it looks like:


<img width="931" alt="Screenshot 2024-06-28 at 22 20 44" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/43378371/f1b781f6-05cc-4f41-94a1-6c7a135cc3cd">



The issue is related to configuring the Pod target when compiling the iOS project which could eventually be pointing to `installer.pods_project.targets.each`
 
**How to solve it?** There are a few steps you can check within your project:

1. Start with updating your project React Native version to 0.74.2 e.g. `Flipper` is disabled with the latest version and it is not compatible with `use_frameworks!` anymore. 
2. Worth trying to build the project with Xcode 15.4 as well for an easy start to see if the version you are using does not get along with the project React Native version.
3. Make sure `use_react_native` and `react_native_post_install` are properly set in the `Podfile`. Feels like all of the required project dependencies are not loaded which may cause this issue. To let you know, `use_react_native` is for dependencies provided by RN itself and `react_native_post_install` is for all the project configurations as a final step.
4. Try it with a new React Native project. If it is working ok with the newly created project, then it’s a hint that the issue is in the local configuration of the Podfile or the project. Compare the differences in the Podfile and check the pods that are installed with the new and old projects. Then track down them.
5. You can also try to `[patch-package](https://www.npmjs.com/package/patch-package)` to create a patch.
7.  Another thing you can try is to set the pod target’s header in the `post_install` hook by adding the following lines inside it
```
 installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings["HEADER_SEARCH_PATHS"] ||= "$(inherited) "
        config.build_settings["HEADER_SEARCH_PATHS"] << '"$(PODS_TARGET_SRCROOT)" '
      end
    end
```

I will close the thread for now, you can re-open it by replying to it.
